### PR TITLE
🌱 Update dependencies except k8s.io/*, helm.sh and gomega

### PR DESCRIPTION
**Pull request by oscr**
**What this PR does / why we need it**:

As discussed in #7490 [comment](https://github.com/kubernetes-sigs/cluster-api/pull/7490#issuecomment-1303539082) here is a pr that updates the other dependencies. I used the following command to avoid updating k8s.io/* helm.sh (since it automatically bumped k8s.io dependencies) and gomega (since it breaks).

`cat go.mod | grep -vE "indirect|\)|\(|replace|go|module|k8s.io|helm.sh|gomega|^$" | awk '{print $1;}' | xargs -I % go get -u '%'`

EDIT: Note it's two commits since it's based on #7490 which isn't merged yet.

cc @sbueringer 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -67,9 +67,9 @@ require (
 	github.com/coreos/go-semver v0.3.0 // indirect
 	github.com/coreos/go-systemd v0.0.0-20191104093116-d3cd4ed1dbcf // indirect
 	github.com/coreos/go-systemd/v22 v22.3.2 // indirect
-	github.com/evanphx/json-patch v4.12.0+incompatible
+	github.com/evanphx/json-patch v5.6.0+incompatible
```
Comment (sbueringer) [inline go.mod:70]: That one is probably a false positive based on the way you bumped. 

We already use: `github.com/evanphx/json-patch/v5 v5.6.0`

Can you try to use the v5 import path in cluster/topology.go? This way we should be able to get rid of the v4 dependency entirely

File: hack/tools/go.mod [hunk 2]
```diff
@@ -51,16 +51,16 @@ require (
...
-	github.com/docker/docker v20.10.17+incompatible // indirect
+	github.com/docker/docker v20.10.21+incompatible // indirect
 	github.com/docker/docker-credential-helpers v0.6.4 // indirect
 	github.com/docker/go-connections v0.4.0 // indirect
 	github.com/docker/go-metrics v0.0.1 // indirect
 	github.com/docker/go-units v0.4.0 // indirect
 	github.com/drone/envsubst/v2 v2.0.0-20210730161058-179042472c46 // indirect
 	github.com/emicklei/go-restful/v3 v3.9.0 // indirect
-	github.com/evanphx/json-patch v4.12.0+incompatible // indirect
+	github.com/evanphx/json-patch v5.6.0+incompatible // indirect
```
Comment (sbueringer) [inline hack/tools/go.mod:61]: Same here

File: test/go.mod [hunk 3]
```diff
@@ -52,9 +52,9 @@ require (
 	github.com/docker/go-units v0.4.0 // indirect
 	github.com/drone/envsubst/v2 v2.0.0-20210730161058-179042472c46 // indirect
 	github.com/emicklei/go-restful/v3 v3.9.0 // indirect
-	github.com/evanphx/json-patch v4.12.0+incompatible // indirect
+	github.com/evanphx/json-patch v5.6.0+incompatible // indirect
```
Comment (sbueringer) [inline test/go.mod:55]: Same

**Comment by sbueringer** [comment]
Thx!

File: go.mod [hunk 4]
```diff
@@ -19,17 +19,17 @@ require (
...
-	github.com/onsi/gomega v1.20.1
+	github.com/mattn/go-runewidth v0.0.14 // indirect
+	github.com/onsi/ginkgo/v2 v2.4.0
+	github.com/onsi/gomega v1.22.1
 	github.com/pkg/errors v0.9.1
-	github.com/spf13/cobra v1.5.0
+	github.com/spf13/cobra v1.6.1
 	github.com/spf13/pflag v1.0.5
-	github.com/spf13/viper v1.12.0
+	github.com/spf13/viper v1.13.0
```
Comment (sbueringer) [inline go.mod:28]: @Cecile what do you think about this PR? (only the second commit)

It should come down to only bumping cobra & viper a minor version

Comment (sbueringer) [reply go.mod:28]: Release notes of both look good to me

**Comment by sbueringer** [comment]
Thx!

/lgtm

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Pull request merged**